File: GraduateThesisManager/GraduateThesisManager.Web/Modules/Thesis/ThesisRegistration/ThesisRegistrationForm.cs

using GraduateThesisManager.Administration.Entities;
using GraduateThesisManager.Common.Entities;
using GraduateThesisManager.Thesis.Entities;

namespace GraduateThesisManager.Thesis.Forms
{
    using Serenity;
    using Serenity.ComponentModel;
    using Serenity.Data;
    using System;
    using System.ComponentModel;
    using System.Collections.Generic;
    using System.IO;

    [FormScript("Thesis.ThesisRegistration")]
    [BasedOnRow(typeof(Entities.ThesisRegistrationRow), CheckNames = true)]
    public class ThesisRegistrationForm
    {
        [Category("Registration Information")]
        [Hidden]
        [LookupEditor(typeof(UserRow))]
        public Int32 User { get; set; }
        [ReadOnly(true)]
        public String UserDisplayName { get; set; }

        [Hidden]
        [LookupEditor(typeof(ThesisPoolRow))]
        public Int32 Thesis { get; set; }
        [ReadOnly(true)]
        public String ThesisThesisName { get; set; }

        [LookupEditor(typeof(RegistrationStatusRow))]
        public Int32 Status { get; set; }
        [DateTimeEditor(IntervalMinutes = 1)]
        [ReadOnly(true)]
        public DateTime CreateDate { get; set; }
        [DateTimeEditor(IntervalMinutes = 1)]
        [ReadOnly(true)]
        public DateTime ExpireDate { get; set; }
        [ReadOnly(true)]
        public String LecturerName { get; set; }
    }
}

File: GraduateThesisManager/GraduateThesisManager.Web/Imports/ServerTypings/Thesis.RegistrationForm.ts
namespace GraduateThesisManager.Thesis {
    export interface RegistrationForm {
        User: Serenity.IntegerEditor;
        Thesis: Serenity.IntegerEditor;
        Status: Serenity.IntegerEditor;
        CreateDate: Serenity.DateTimeEditor;
        ExpireDate: Serenity.DateTimeEditor;
        UserDisplayName: Serenity.StringEditor;
        ThesisThesisName: Serenity.StringEditor;
    }

    export class RegistrationForm extends Serenity.PrefixedContext {
        static formKey = 'Thesis.Registration';
        private static init: boolean;

        constructor(prefix: string) {
            super(prefix);

            if (!RegistrationForm.init)  {
                RegistrationForm.init = true;

                var s = Serenity;
                var w0 = s.IntegerEditor;
                var w1 = s.DateTimeEditor;
                var w2 = s.StringEditor;

                Q.initFormType(RegistrationForm, [
                    'User', w0,
                    'Thesis', w0,
                    'Status', w0,
                    'CreateDate', w1,
                    'ExpireDate', w1,
                    'UserDisplayName', w2,
                    'ThesisThesisName', w2
                ]);
            }
        }
    }
}

